### PR TITLE
Rename all instances of PlayerAvatar, FieldSystem

### PR DESCRIPTION
**Pull request by adrienntindall**
This took too long

### CODE REVIEW

File: src/comm_player_manager.c [hunk 1]
```diff
@@ -388,48 +388,46 @@ static void CommPlayer_Add (u8 netId)
...
+                MapObject * obj = MapObjMan_LocalMapObjByIndex(sCommPlayerManager->fieldSystem->mapObjMan, 0xff + netId + 1);
 
-                    if (obj != 0) {
-                        MapObject_Delete(obj);
-                    }
+                if (obj != NULL) {
+                    MapObject_Delete(obj);
                 }
             }
+        }
```
Comment (lhearachel) [inline src/comm_player_manager.c:402]: Might be more readable to condense these `if` chains to a single compound:

```c
if (trainerInfo
        && !sCommPlayerManager->isUnderground
        && netId != CommSys_CurNetId()) {
    MapObject * obj = MapObjMan_LocalMapObjByIndex(sCommPlayerManager->fieldSystem->mapObjMan, 0xff + netId + 1);
    
    if (obj != NULL) {
        MapObject_Delete(obj);
    }
}
```

File: src/comm_player_manager.c [hunk 2]
```diff
@@ -388,48 +388,46 @@ static void CommPlayer_Add (u8 netId)
...
 
-            {
-                int version = 0;
-
-                if (0 == TrainerInfo_GameCode(trainerInfo)) {
-                    version = 1;
-                }
+        {
+            int version = 0;
```
Comment (lhearachel) [inline src/comm_player_manager.c:405]: Is this block necessary to match?

File: src/field_comm_manager.c [hunk 3]
```diff
@@ -73,20 +73,20 @@ static void sub_02059658 (void)
...
 
-void FieldCommMan_Init (FieldSystem * fieldSys)
+void FieldCommMan_Init (FieldSystem * fieldSystem)
 {
     if (sFieldCommMan != NULL) {
         return;
     }
 
-    CommFieldCmd_Init((void *)fieldSys);
+    CommFieldCmd_Init((void *)fieldSystem);
```
Comment (lhearachel) [inline src/field_comm_manager.c:82]: Is this cast necessary?

File: src/map_object_move.c [hunk 4]
```diff
@@ -853,55 +853,56 @@ u32 sub_02064238 (MapObject * mapObj, int param1)
...
+    pos.y += vec->y;
+    pos.z += vec->z;
 
-    MapObject_SetPosVec(mapObj, &v0);
+    MapObject_SetPosVec(mapObj, &pos);
 }
 
-void sub_020642A4 (MapObject * mapObj, int param1, fx32 param2)
+//TODO: This could really use a better name
+void MapObject_AddValInDir (MapObject * mapObj, int dir, fx32 param2)
```
Comment (lhearachel) [inline src/map_object_move.c:876]: `MapObject_StepDirection`?

File: src/map_object_move.c [hunk 5]
```diff
@@ -933,40 +934,40 @@ int sub_02064390 (MapObject * mapObj)
...
         return 0;
     }
 
-    MapObject_SetStatusFlagOff(mapObj, (1 << 11));
+    MapObject_SetStatusFlagOff(mapObj, MAP_OBJ_STATUS_11);
     return 1;
 }
 
-void sub_02064418 (int param0, VecFx32 * param1, fx32 param2)
+void VecFx32_AddValInDirection (int dir, VecFx32 * vec, fx32 val)
```
Comment (lhearachel) [inline src/map_object_move.c:957]: I think this one deserves whatever suffix we decide on for the similar `MapObject` function above.

File: src/map_object_move.c [hunk 6]
```diff
@@ -987,32 +988,32 @@ void sub_02064464 (MapObject * mapObj)
...
-    0x3,
-    0x2
+    DIR_SOUTH,
+    DIR_NORTH,
+    DIR_EAST,
+    DIR_WEST
 };
 
-int sub_0206447C (int param0)
+int Direction_GetOpposite (int dir)
```
Comment (lhearachel) [inline src/map_object_move.c:997]: `Direction_Flip`? Or maybe just `FlipDirection`?

File: src/overlay005/ov5_021D1C30.c [hunk 7]
```diff
@@ -359,20 +359,20 @@ int ov5_021D1DA4 (const UnkStruct_ov5_021D1CAC * param0, FieldSystem * fieldSyst
...
 
-    if (sub_0203CBE0(param0, &v0) == 1) {
-        sub_0203E880(param0, sub_02062960(v0), v0);
+    if (sub_0203CBE0(fieldSystem, &v0) == 1) {
+        sub_0203E880(fieldSystem, sub_02062960(v0), v0);
         return 1;
     }
 
-    v1 = sub_0203CB80(param0, (void *)sub_0203A440(param0), sub_0203A448(param0));
+    v1 = sub_0203CB80(fieldSystem, (void *)sub_0203A440(fieldSystem), sub_0203A448(fieldSystem));
```
Comment (lhearachel) [inline src/overlay005/ov5_021D1C30.c:372]: Is this void-cast necessary?

File: src/comm_player_manager.c (hunk 1 above)
Comment (adrienntindall) [reply src/comm_player_manager.c:402]: that obviously doen't match though? the second two can be condensed but ``trainerInfo`` has more in its block

File: src/field_comm_manager.c (hunk 3 above)
Comment (adrienntindall) [reply src/field_comm_manager.c:82]: nope, I'll start removing them as I see them lol

File: src/map_object_move.c (hunk 4 above)
Comment (adrienntindall) [reply src/map_object_move.c:876]: That's better yeah

File: src/map_object_move.c (hunk 6 above)
Comment (adrienntindall) [reply src/map_object_move.c:997]: Opposite is fine tbh

File: src/comm_player_manager.c (hunk 1 above)
Comment (lhearachel) [reply src/comm_player_manager.c]: Excess scope blocks got me confused... at least merge the ones that can be merged. Fewer indentation levels are good.

**Review by lhearachel — APPROVED** [review]

**Pull request merged**